docs/griptape-framework/drivers/vector-store-drivers.md

```diff
@@ -406,3 +406,112 @@ vector_store_driver.upsert_text_artifacts(
+from griptape.loaders import PdfLoader
+from griptape.chunkers import TextChunker
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+
+# Setting the models
+embedding_model = "sentence-transformers/all-MiniLM-L6-v2"
+qdrant_model = SentenceTransformer("all-MiniLM-L6-v2")
+file_name = "linux_bible.pdf"
+HUGGINGFACE_TOKEN = os.environ["HUGGINGFACE_HUB_ACCESS_TOKEN"]
+
+
+# Using HuggingFaceTokenizer
+def create_tokenizer(embedding_model):
+    tokenizer = HuggingFaceTokenizer(max_output_tokens=1024, tokenizer=AutoTokenizer.from_pretrained(embedding_model))
+    return tokenizer
+
+
+def create_embedding_driver(embedding_model, tokenizer):
+    embedding_driver = HuggingFaceHubEmbeddingDriver(
+        api_token=HUGGINGFACE_TOKEN, model=embedding_model, tokenizer=tokenizer
+    )
+    return embedding_driver
+
+
+# Instantiating QdrantVectorStoreDriver
+def create_vector_store_driver(url, collection_name, embedding_driver):
+    vector_store_driver = QdrantVectorStoreDriver(
+        url=url, collection_name=collection_name, content_payload_key="content", embedding_driver=embedding_driver
+    )
+    return vector_store_driver
+
+
+# Opening the file
+def load_pdf(file_name, tokenizer):
+    with open(file_name, "rb") as f:
+        loader = PdfLoader(tokenizer=tokenizer, chunker=TextChunker(tokenizer=tokenizer, max_tokens=1024)).load(
+            f.read()
+        )
+    return loader
+
+
+def main():
+    tokenizer = create_tokenizer(embedding_model)
+    embedding_driver = create_embedding_driver(embedding_model, tokenizer)
+    vector_store_driver = create_vector_store_driver(
+        url="http://localhost:6333", collection_name="linux_bible", embedding_driver=embedding_driver
+    )
+
+    # Loading the data
+    loader = load_pdf(file_name, tokenizer=tokenizer)
+
+    # Generate metadata for each chunk (example metadata)
+    metadata = [{"source": file_name, "page": i + 1} for i in range(len(loader))]
+
+    # Upserting the vector
+    try:
+        for i, l in enumerate(loader):
+            content = str(l)
+            meta = metadata[i] if metadata else None
+            vector_store_driver.upsert_vector(embedding_driver.try_embed_chunk(content), meta=meta, content=content)
+        logging.info("Successfully upserted vectors with metadata.")
+    except Exception as e:
+        logging.error(f"Error during upsert_vector: {e}")
+
+    # Querying the data
+    query_string = "Who created linux?"
+    query_results = vector_store_driver.query(query_string, count=6, include_vectors=True)
+    for result in query_results:
+       print(f"ID: {result.id}, Score: {result.score}, Vector: {result.vector}, Metadata: {result.meta}")
+
+    # Retrieving single entries
+    single_entry = vector_store_driver.load_entry(vector_id="00499354-9362-49c7-97b6-0220b9de84e7")
+    if single_entry:
+        print(f"Vector ID: {single_entry.id}")
+        print(f"Vector: {single_entry.vector}")
+        print(f"Metadata: {single_entry.meta}")
```

Comment:
try to avoid a lot of these intermediary `print`s. using [some other snippets](https://github.com/griptape-ai/griptape/blob/dev/docs/griptape-framework/drivers/vector-store-drivers.md) as an example, try to only print out the end result.